drools-reliability/drools-reliability-core/src/main/java/org/drools/reliability/core/SimpleSerializationReliableObjectStore.java

```diff
@@ -39,7 +42,17 @@ public SimpleSerializationReliableObjectStore() {
 
     public SimpleSerializationReliableObjectStore(Storage<Long, StoredObject> storage) {
         super();
-        this.storage = storage;
+        this.storage = storage.size()>0 ? updateObjectReferences(storage) : storage;
+    }
+
+    private Storage<Long, StoredObject> updateObjectReferences(Storage<Long, StoredObject> storage){
+        Storage<Long, StoredObject> updateStorage = storage;
+
+        for (Long key: storage.keySet()){
+            updateStorage.put(key, ((SerializableStoredObject) storage.get(key)).updateReferencedObjects(storage));
```

Comment:
```suggestion
            if (storage.get(key) instanceof SerializableStoredObject) {
                updateStorage.put(key, ((SerializableStoredObject) storage.get(key)).updateReferencedObjects(storage));
            }
```

The cast to `SerializableStoredObject` causes `ClassCastException` in ReliabilityCepTimedOutTest etc. because `SerializableStoredEvent` is not a sub-class of  `SerializableStoredObject` It's a little hard to change the inheritance structure because it's a kind of multi-inheritance. I'll review in a separate JIRA.

At the moment, we skip the feature for Event so that tests can pass.